interface/src/hooks/useCalendarsAndEvents.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { createContext, useContext, useEffect, useState } from "react"
+import UrbitApi from "../logic/api"
+import Event, { EventForm } from "../types/Event"
+import Calendar, { DEFAULT_PERMISSIONS } from "../types/Calendar"
+import { CalendarViewState } from "../views/CalendarView"
+
+type CalendarAndEventContextType = {
+  events: Event[]
+  setEvents: (arg: Event[]) => void
+  getEvents?: () => Promise<void>
+  deleteEvent?: (event: Event) => void
+  saveEvent?: (event: EventForm, update: boolean) => void
+  calendars: Calendar[]
+  setCalendars: (arg: Calendar[]) => void
+  getCalendars?: () => Promise<void>
+  saveCalendar?: (data: CalendarViewState, update: boolean) => Promise<void>
+  deleteCalendar?: (calendar: Calendar) => Promise<boolean>
+  toggleCalendar?: (calendar: Calendar) => void
+}
+
+export const CalendarAndEventContext =
+  createContext<CalendarAndEventContextType>({
+    events: [],
+    setEvents: () => {},
+    calendars: [],
+    setCalendars: () => {},
+  })
+
+export const CalendarAndEventProvider: React.FC = ({ children }) => {
+  const api = new UrbitApi()
+  const [events, setEvents] = useState<Event[] | undefined>(undefined)
```

Comment:
My personal preference is to use an empty array as the null state, rather than `undefined`. It avoids errors being thrown for array methods when the expected behavior is to do nothing for an empty array anyway. (like a map to created a list of components)

Reply:
Sure, I can switch this out.